modin/experimental/core/execution/native/implementations/omnisci_on_native/exchange/dataframe_protocol/column.py

```diff
@@ -0,0 +1,488 @@
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""The module houses OmnisciOnNative implementation of the Column class of DataFrame exchange protocol."""
+
+import pyarrow as pa
+import pandas
+import numpy as np
+
+from typing import Any, Optional, Tuple, Dict, Iterable
+from math import ceil
+from modin.core.dataframe.base.exchange.dataframe_protocol.utils import (
+    DTypeKind,
+    ColumnNullType,
+    pandas_dtype_to_arrow_c,
+)
+from modin.core.dataframe.base.exchange.dataframe_protocol.dataframe import (
+    ProtocolColumn,
+)
+from modin.utils import _inherit_docstrings
+from .buffer import OmnisciProtocolBuffer
+from .utils import arrow_dtype_to_arrow_c
+
+
+@_inherit_docstrings(ProtocolColumn)
+class OmnisciProtocolColumn(ProtocolColumn):
+    """
+    Wrapper of ``OmnisciProtocolDataframe`` holding a single column.
+
+    The Column object wraps a ``ProtocolDataframe`` to ease referencing original
+    Modin DataFrame with no materialization of PyArrow table where possible.
+    ``ProtocolDataframe`` also already implements methods like chunking and ``allow_copy``
+    checks, so we can just forward calls for the methods to ``ProtocolDataFrame`` without
+    reimplementing them.
+
+    Parameters
+    ----------
+    column : OmnisciProtocolDataframe
+        DataFrame protocol object holding a PyArrow table with a single column.
+
+    Notes
+    -----
+    The object could be modified inplace due to casting PyArrow buffers to a new dtype:
+    ``_propagate_dtype``, ``_cast_at`` - the methods replace the wrapped
+    ``OmnisciProtocolDataframe`` object with the new one holding the casted PyArrow table.
+    """
+
+    def __init__(self, column: "OmnisciProtocolDataframe") -> None:
+        self._col = column
+
+    @property
+    def size(self) -> int:
+        return self._col.num_rows()
+
+    @property
+    def offset(self) -> int:
+        # The offset may change if it would require to cast buffers as the casted ones
+        # no longer depend on their parent tables. So materializing buffers
+        # before returning the offset
+        self._materialize_actual_buffers()
+        return self._pyarrow_table.column(0).chunks[0].offset
+
+    @property
+    def dtype(self) -> Tuple[DTypeKind, int, str, str]:
+        dtype = self._pandas_dtype
+
+        if pandas.api.types.is_bool_dtype(dtype):
+            return (DTypeKind.BOOL, 1, pandas_dtype_to_arrow_c(np.dtype("bool")), "=")
```

Comment:
We should probably allow this case to go in _dtype_from_primitive_pandas. Also, the bitmask for bool is not equal to 1.

Reply:
Arrow stores boolean columns as a bit-mask, so we can't handle this in `_from_primitive_pandas` as it assignes `bitwidth=8` for bools
```python
>>> pyarrow.Table.from_pydict({"a": [True, False, True]}).column(0).type.bit_width
1
```

Reply:
Ok, I see.